app/src/main/java/com/example/android/navigation/GameWonFragment.kt

```diff
@@ -37,10 +38,36 @@ class GameWonFragment : Fragment() {
             view?.findNavController()?.navigate(GameWonFragmentDirections.actionGameWonFragmentToGameFragment())
         }
         displayArgs()
+        setHasOptionsMenu(true)
         return binding.root
     }
-    private fun displayArgs(){
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.winner_menu, menu)
+        if (null == getShareIntent().resolveActivity(activity!!.packageManager)) {
+            menu.findItem(R.id.share)?.isVisible = false
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.share -> shareSuccess()
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun displayArgs() {
         val args = GameWonFragmentArgs.fromBundle(arguments!!)
         Toast.makeText(context, "NumCorrect: ${args.numCorrect}, NumQuestions: ${args.numQuestions}", Toast.LENGTH_LONG).show()
     }
+
+    private fun getShareIntent(): Intent {
+        val args = GameWonFragmentArgs.fromBundle(arguments!!)
```

Comment:
We should avoid using !! as much as possible. Find out more about why is a bad practice to use !! and try to refactor your code by not using this.

Reply:
(!!) - the null operator ensures that the value of the variable is not null, and if it is null throw a null pointer exception. 
It should be used carrefully because it force-casts a nullable variable to a non-null variable.